src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,11 +376,219 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             }
         }
 
+        private ImmutableArray<BoundExpression> VisitArguments(
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            ImmutableArray<int> argsToParamsOpt,
+            ref ImmutableArray<RefKind> argumentRefKindsOpt,
+            [NotNullIfNotNull("visitedReceiver")] ref BoundExpression? rewrittenReceiver,
+            out ArrayBuilder<LocalSymbol>? temps,
+            out BitVector positionsAssignedToTemp,
+            out bool receiverIsArgumentSideEffectSequence)
+        {
+            Debug.Assert(argumentRefKindsOpt.IsDefault || argumentRefKindsOpt.Length == arguments.Length);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor };
+            Debug.Assert(!requiresInstanceReceiver || rewrittenReceiver != null || _inExpressionLambda);
+            temps = null;
+            positionsAssignedToTemp = BitVector.Null;
+            receiverIsArgumentSideEffectSequence = false;
+
+            if (arguments.IsEmpty)
+            {
+                return arguments;
+            }
+
+            var visitedArgumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            ArrayBuilder<BoundExpression>? sideEffectsBuilder = null;
+            int lastIndexStored = -1;
+            bool receiverStored = false;
+            ArrayBuilder<RefKind>? refKindsBuilder = null;
+            var parameters = methodOrIndexer.GetParameters();
+
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                InterpolatedStringHandlerData? nullableInterpolatedData = setupInterpolationDataIfNeeded(
```

Comment:
>nullableInterpolatedData [](http://example.com/codeflow?start=47&length=24)

It doesn't look like we need the entire ```InterpolatedStringHandlerData``` back. An array of placeholders to remove would be sufficient, in my opinion.